test/functional/http/post/6.dapps.inTransfer.js

```diff
@@ -0,0 +1,233 @@
+					badTransactions.push(transaction);
+				});
+			});
+
+			it('with number should fail', function () {
+				transaction = node.lisk.transfer.createInTransfer(node.guestbookDapp.id, Date.now(), node.gAccount.password);
+				transaction.asset.inTransfer.dappId = 1.2;
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('status').to.equal(400);
+					node.expect(res).to.have.nested.property('body.message').to.equal('Invalid transaction body - Failed to validate inTransfer schema: Expected type string but found type number');
+					badTransactions.push(transaction);
+				});
+			});
+
+			it('with empty array should fail', function () {
+				transaction = node.lisk.transfer.createInTransfer(node.guestbookDapp.id, Date.now(), node.gAccount.password);
+				transaction.asset.inTransfer.dappId = [];
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('status').to.equal(400);
+					node.expect(res).to.have.nested.property('body.message').to.equal('Invalid transaction body - Failed to validate inTransfer schema: Expected type string but found type array');
+					badTransactions.push(transaction);
+				});
+			});
+
+			it('with empty object should fail', function () {
+				transaction = node.lisk.transfer.createInTransfer(node.guestbookDapp.id, Date.now(), node.gAccount.password);
+				transaction.asset.inTransfer.dappId = {};
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('status').to.equal(400);
+					node.expect(res).to.have.nested.property('body.message').to.equal('Invalid transaction body - Failed to validate inTransfer schema: Expected type string but found type object');
+					badTransactions.push(transaction);
+				});
+			});
+
+			it('invalid dapp id should fail', function () {
+				var invalidDappId = '1L';
+				transaction = node.lisk.transfer.createInTransfer(invalidDappId, 1, node.gAccount.password);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('status').to.equal(400);
+					node.expect(res).to.have.nested.property('body.message').to.equal('Invalid transaction body - Failed to validate inTransfer schema: Object didn\'t pass validation for format id: ' + invalidDappId);
+					badTransactions.push(transaction);
+				});
+			});
+		});
+
+		describe('amount', function () {
+
+			it('using < 0 should fail', function () {
+				transaction = node.lisk.transfer.createInTransfer(node.guestbookDapp.id, -1, node.gAccount.password);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('status').to.equal(400);
+					node.expect(res).to.have.nested.property('body.message').to.equal('Invalid transaction body - Failed to validate transaction schema: Value -1 is less than minimum 0');
```

Comment:
Why are we allowing 0 lsk in transfers? The minimum should be 1, just like a type 0.

Reply:
This the minimum defined by the parent transaction schema, where child types might only have a fee.

Then for an actual `transfer` it is checked here in verify: https://github.com/LiskHQ/lisk/blob/development/logic/transfer.js#L62

Schema checks are designed to be less concrete to allow for inheritance pattern.